pyspecProcScripts/integral_w_error.py

```diff
@@ -1,46 +1,63 @@
 import pyspecdata as psp
 from .integrate_limits import integrate_limits
 from .simple_functions import select_pathway
+import pyspecProcScripts as psdpr
 import logging
 import numpy as np
 
 
+def masked_mean_multi(x, axis=None):
+    "Calculates the mean on a 1D axis"
+    assert axis is not None
+
+    def masked_mean(x):
+        return np.mean(x[np.isfinite(x)])
+
+    return np.apply_along_axis(masked_mean, axis, x)
+
+
+def masked_var_multi(x, var_has_imag=False, axis=None):
+    assert axis is not None
+
+    def masked_var(x):
+        if var_has_imag:
+            return np.var(x[np.isfinite(x)], ddof=1) / 2
+        else:
+            return np.var(x[np.isfinite(x)], ddof=1)
+
+    return np.apply_along_axis(masked_var, axis, x)
+
+
```

Comment:
In JF words: "I know how to write a masked mean or std only along 1 dimensions, so use numpy apply_along_axis to make it a function that works along 1 dimension of multidimensional data" (pulled from our scripts in t_domain)